### PR TITLE
Prac 11 registration form frontend

### PR DESCRIPTION
**Pull request by ElenaDu**
## One Line Description
UI with basic functionality for Registration Form with input fields validation.
## Requirements

See Figma for design details. The registration form allows a user to register. It performs field validation and displays errors when necessary.
## Notes
Please note that this pull request is for the registration form, and the login form will be added separately.

## Test Steps

1. Check the registration process when all the fields are filled correctly.
2. Verify that the Register button is unavailable until all the fields are filled.
3. Check error messages for the input fields.

## UI/UX Outcome

Screen Shot 2024-03-29 at 5 14 26 PM

Screen Shot 2024-03-29 at 5 13 41 PM

Screen Shot 2024-03-29 at 4 39 26 PM

### CODE REVIEW

File: src/components/RegistrationForm.tsx [hunk 1]
```diff
@@ -21,81 +26,260 @@ const labelStyle = {
...
-                    <Stack direction="row" display="flex" alignItems="center">
-                        <Radio value="parent" flex="1" bg="white" size="md">
-                            <span style={radioLabelStyle}>Parent</span>
-                        </Radio>
-                        <Radio value="tutor" flex="1" bg="white" size="md">
-                            <span style={radioLabelStyle}>Tutor</span>
-                        </Radio>
+        <Formik
+            initialValues={initialValues}
+            validationSchema={Yup.object({
```
Comment (IrynaKolh) [inline src/components/RegistrationForm.tsx]: In file structure there is folder "validationSchemas" move this validation object to index.ts file.

File: src/components/RegistrationForm.tsx [hunk 2]
```diff
@@ -21,81 +26,260 @@ const labelStyle = {
...
+                        email: values.email,
+                        password: values.password,
+                        role: values.role,
+                    };
+                    //console.log('Request Data:', requestData);
+                    const response = await axios.post(
+                        'https://ffprac-team2-back.onrender.com/api/v1/auth/register',
+                        requestData
+                    );
+                    const { firstName, lastName, email, role, token } = response.data.user;
```
Comment (IrynaKolh) [inline src/components/RegistrationForm.tsx]: response.data has a object 
 {
    "user": {
        "firstName": "Some name",
        "lastName": "some lasr name",
        "email": "someEmail@gmail.com",
        "role": "some role"
    },
    "token": "you token"
}
check do you realy get token in user?

**Review by IrynaKolh — COMMENTED** [review]
check do you realy get token in user?

File: src/components/RegistrationForm.tsx [hunk 3]
```diff
@@ -21,81 +26,260 @@ const labelStyle = {
...
+                    const requestData = {
+                        firstName: values.firstName,
+                        lastName: values.lastName,
+                        email: values.email,
+                        password: values.password,
+                        role: values.role,
+                    };
+                    //console.log('Request Data:', requestData);
+                    const response = await axios.post(
+                        'https://ffprac-team2-back.onrender.com/api/v1/auth/register',
```
Comment (oxangyal) [inline src/components/RegistrationForm.tsx]: For best practices I would recommend to hide the full URL used for making requests in the .env file. This guarantees a more secure management of confidential data such as API URLs and helps the process of changing URLs without modifying the application code. I would use something like REACT_APP_BASE_URL for instance.

Comment (ElenaDu) [reply src/components/RegistrationForm.tsx]: Done

File: src/components/RegistrationForm.tsx (hunk 2 above)
Comment (ElenaDu) [reply src/components/RegistrationForm.tsx]: Done

File: src/components/RegistrationForm.tsx (hunk 1 above)
Comment (ElenaDu) [reply src/components/RegistrationForm.tsx]: Done

**Review by IrynaKolh — APPROVED** [review]
LGTM

**Review by oxangyal — APPROVED** [review]
LGTM

**Pull request merged**